src/Integration/State/StateManager.cs

```diff
@@ -176,6 +196,14 @@ private void SetProjectsUIThread(ConnectionInformation connection, IEnumerable<P
             }
         }
 
+        private void DisposeConnections()
+        {
+            this.ManagedState.ConnectedServers
```

Comment:
TransferableVisualState - don't think it needs to be disposable it's just a thin container over the VM (which internally have disposable object). That class is not responsible for the lifecycle management of the objects it exposes, the manager is, and the manager is IDisposable.
It is true that DisposeConnections needs to be called when the state manager is disposed and it's not at the moment (will fix it and see if I can add a test).